cdci_data_analysis/analysis/drupal_helper.py

```diff
@@ -0,0 +1,367 @@
+                parsed_t1 = parser.parse(t1)
+                parsed_t2 = parser.parse(t2)
+                t_start = parser.parse(times[0])
+                t_end = parser.parse(times[1])
+                if t_start == parsed_t1 and t_end == parsed_t2:
+                    observation_drupal_id = observation['nid']
+                    break
+
+        if observation_drupal_id is None:
+            observation_drupal_id = post_observation(product_gallery_url, jwt_token, t1, t2)
+
+    return observation_drupal_id
+
+
+def post_data_product_to_gallery(product_gallery_url, session_id, job_id, jwt_token,
+                                 product_title=None,
+                                 img_fid=None,
+                                 observation_id=None,
+                                 user_id_product_creator=None,
+                                 **kwargs):
+    # body_gallery_article_node = body_article_product_gallery.body_article.copy()
+    body_gallery_article_node = copy.deepcopy(body_article_product_gallery.body_article)
+
+    # set the type of content to post
+    link_content_type = body_gallery_article_node["_links"]["type"]["href"] + 'data_product'
+    body_gallery_article_node["_links"]["type"]["href"] = link_content_type
+
+    # set the initial body content
+    body_value = ''
+    product_type = ''
+    t1 = t2 = None
+    # get products
+    scratch_dir_json_fn = f'scratch_sid_{session_id}_jid_{job_id}'
+    # the aliased version might have been created
+    scratch_dir_json_fn_aliased = f'scratch_sid_{session_id}_jid_{job_id}_aliased'
+    analysis_parameters_json_content_original = None
+    #
+    if os.path.exists(scratch_dir_json_fn):
+        analysis_parameters_json_content_original = json.load(open(scratch_dir_json_fn + '/analysis_parameters.json'))
+    elif os.path.exists(scratch_dir_json_fn_aliased):
+        analysis_parameters_json_content_original = json.load(
+            open(scratch_dir_json_fn_aliased + '/analysis_parameters.json'))
+
+    if analysis_parameters_json_content_original is not None:
+        analysis_parameters_json_content_original.pop('token', None)
+        instrument = analysis_parameters_json_content_original.pop('instrument')
+        product_type = analysis_parameters_json_content_original.pop('product_type')
+        # time data for the observation
+        t1 = analysis_parameters_json_content_original.pop('T1')
+        t2 = analysis_parameters_json_content_original.pop('T2')
+
+        # TODO no need to set all the parameters by default
+        # for k, v in analysis_parameters_json_content_original.items():
+        #     # assuming the name of the field in drupal starts always with field_
+        #     field_name = str.lower('field_' + k)
+        #     body_gallery_article_node[field_name] = [{
+        #         "value": v
+        #     }]
+        body_value = (f'''''')
+    else:
+        raise RequestNotUnderstood(message="Request data ont found",
```

Comment:
`not found`

Reply:
could it be something more descriptive, like "Results of the ODA product request could not be found. Perhaps wrong job_id was passed?". Else we have plenty of very different `requests` around: request as sent now, request to drupal, etc.